phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDataType.java

```diff
@@ -1144,17 +1144,18 @@ public static PDataType fromLiteral(Object value) {
       return null;
     }
     for (PDataType type : PDataType.values()) {
-      if (type.isArrayType()) {
-        PhoenixArray arr = (PhoenixArray) value;
-        if ((type.getSqlType() == arr.baseType.sqlType + PDataType.ARRAY_TYPE_BASE)
-            && type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      } else {
-        if (type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      }
+		if (type.isArrayType() && type.getJavaClass().isInstance(value)) {
```

Comment:
@twdsilva , your placement of isInstance() is good since in current case if it is of Array type but not compatible class then it will call isInstance() twice (once in if () + once in else part )  whereas in your case it will be called only once.